test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Options/PackageSourceMappingPageTests.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Utilities.UnifiedSettings;
+using Moq;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio.Options;
+using NuGet.VisualStudio;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test.Options
+{
+    [Collection(MockedVS.Collection)]
+    public class PackageSourceMappingPageTests : NuGetExternalSettingsProviderTests<PackageSourceMappingPage>
+    {
+        private IEnumerable<PackageSource> _packageSources;
+
+        public PackageSourceMappingPageTests(GlobalServiceProvider sp)
+        {
+            sp.Reset();
+            NuGetUIThreadHelper.SetCustomJoinableTaskFactory(ThreadHelper.JoinableTaskFactory);
+            _packageSources = Enumerable.Empty<PackageSource>();
+        }
+
+        protected override PackageSourceMappingPage CreateInstance(VSSettings vsSettings)
+        {
+            Mock<IPackageSourceProvider> mockedPackageSourceProvider = new Mock<IPackageSourceProvider>();
+            mockedPackageSourceProvider.Setup(packageSourceProvider => packageSourceProvider.LoadPackageSources())
+                .Returns(_packageSources);
+
+            PackageSourceMappingProvider sourceMappingProvider = new(vsSettings);
+            return new PackageSourceMappingPage(vsSettings, mockedPackageSourceProvider.Object, sourceMappingProvider);
+        }
+
+        [Fact]
+        public void VsSettings_SettingsChanged_RaisesEnumSettingChoicesChanged()
+        {
+            // Arrange
+            bool wasEnumSettingChoicesChangedRaised = false;
+            PackageSourceMappingPage instance = CreateInstance(_vsSettings);
+            instance.EnumSettingChoicesChanged += (s, e) =>
+            {
+                wasEnumSettingChoicesChangedRaised = true;
+            };
+
+            // Act
+            instance.VsSettings_SettingsChanged(this, EventArgs.Empty);
+
+            // Assert
+            wasEnumSettingChoicesChangedRaised.Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task SetValueAsync_PreviousMappingsToNonexistantSources_AddNewMapping_ExistingMappingsAreUnchangedAsync()
```

Comment:
I wonder if we can add some more targeted tests. 
Right now it seems like we only have one that tests any of the source mapping adding logic.

Reply:
I would've expected to see more exhaustive tests, ones that focus on the basic scenarios and then build from there.

Example: Add a mapping to a single source, add a mapping to multiple sources, remove a mapping from a single source, remove mapping from multiple sources.

Reply:
I believe what you're describing is covered by the provider tests: https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceMappingProviderTests.cs

This code is just building a collection to pass into that provider. Maybe you're wanting tests that ensure each possible combination received from USX is passed to the provider?

Reply:
I understand it's forwarding things, but there's a bunch of logic that I was thinking can be tested.